### PR TITLE
Adding recoveryservices swagger for api-version 2021-01-01

### PR DESCRIPTION
**Pull request by amchandn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "parameters": {
+    "subscriptionId": "77777777-b0c6-47a2-b37c-d8e65a629c18",
+    "resourceGroupName": "resGroupFoo",
+    "api-version": "2021-01-01",
```
Comment (RamyasreeChakka) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json:5]: While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.

**Comment by RamyasreeChakka** [comment]
@amchandn From change log, I see your PR addressing below two things...
1. Adding new API version
2. Fixing swagger issue for existing version
For faster and easier reviews, please raise separate PRs for 1 & 2.
For 1, While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes in subsequent commits.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json (hunk 1 above)
Comment (amchandn) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json:5]: @RamyasreeChakka , apart from adding the files for new API-version, I've only modified readme.md files to include newer API version and change the default version in the readme files to the latest version.

**Comment by amchandn** [comment]
> @amchandn From change log, I see your PR addressing below two things...
> 
> 1. Adding new API version
> 2. Fixing swagger issue for existing version
>    For faster and easier reviews, please raise separate PRs for 1 & 2.
>    For 1, While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes in subsequent commits.



> @amchandn From change log, I see your PR addressing below two things...
> 
> 1. Adding new API version
> 2. Fixing swagger issue for existing version
>    For faster and easier reviews, please raise separate PRs for 1 & 2.
>    For 1, While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes in subsequent commits.

@RamyasreeChakka , I have not tried to fix any issue for existing API-version. I've just referenced the latest API-version in the readme.md files.

Reverting the readme.md modifications (except the additions) for now

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json (hunk 1 above)
Comment (amchandn) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/examples/CheckNameAvailability_Available.json:5]: My apologies @RamyasreeChakka for selecting the tickbox - 'update existing version to fix swagger quality issue in s360'

I dont need to select this checkbox and I've unselected it.

Can you please review this PR?

**Comment by amchandn** [comment]
> @amchandn From change log, I see your PR addressing below two things...
> 
> 1. Adding new API version
> 2. Fixing swagger issue for existing version
>    For faster and easier reviews, please raise separate PRs for 1 & 2.
>    For 1, While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes in subsequent commits.

@RamyasreeChakka , I had mistakenly selected the checkbox - 'update existing version to fix swagger quality issue in s360' and I've unselected it now.

My changes only include addition of a new api-version for swagger.

**Comment by chiragg4u** [comment]
While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.

**Comment by amchandn** [comment]
> While adding new version the recommendation is to copy the existing version into new directory structure for first commit and then push new changes for easy review. Please create new branch with this recommendation for faster review.

@chiragg4u 
Please find the new PR created by the process you suggested at -
https://github.com/Azure/azure-rest-api-specs/pull/13206

Requesting your review of the above PR

**Pull request closed without merging**